src/test/java/com/hotels/beeju/extensions/ThriftHiveMetaStoreJUnitExtensionTest.java

```diff
@@ -0,0 +1,117 @@
+
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.metastore.HiveMetaStoreClient;
+import org.apache.hadoop.hive.metastore.api.AlreadyExistsException;
+import org.apache.hadoop.hive.metastore.api.Database;
+import org.apache.hadoop.hive.metastore.api.InvalidObjectException;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import com.hotels.beeju.ThriftHiveMetaStoreJUnitRule;
+import com.hotels.beeju.ThriftHiveMetaStoreJUnitRuleTest;
+
+public class ThriftHiveMetaStoreJUnitExtensionTest implements BeforeEachCallback {
+
+  @RegisterExtension
+  ThriftHiveMetaStoreJUnitExtension hiveDefaultName = new ThriftHiveMetaStoreJUnitExtension();
+
+  @RegisterExtension
+  ThriftHiveMetaStoreJUnitExtension hiveCustomName = new ThriftHiveMetaStoreJUnitExtension("my_test_database");
+
+  private static File defaultTempRoot;
+  private static File customTempRoot;
+
+  @Override
+  public void beforeEach(ExtensionContext context){
+    defaultTempRoot = hiveDefaultName.getTempDirectory();
+    assertTrue(defaultTempRoot.exists());
+    customTempRoot = hiveCustomName.getTempDirectory();
+    assertTrue(customTempRoot.exists());
+  }
+
+  @Test
+  public void hiveDefaultName() throws Exception {
+    assertRuleInitialised(hiveDefaultName);
+  }
+
+  @Test
+  public void hiveCustomName() throws Exception {
+    assertRuleInitialised(hiveCustomName);
+  }
+
+  private static void assertRuleInitialised(ThriftHiveMetaStoreJUnitExtension hive) throws Exception {
+    String databaseName = hive.databaseName();
+
+    Database database = hive.client().getDatabase(databaseName);
+
+    assertThat(database.getName(), is(databaseName));
+    File databaseFolder = new File(hive.getTempDirectory(), databaseName);
+    assertThat(new File(database.getLocationUri()) + "/", is(databaseFolder.toURI().toString()));
+
+    assertThat(hive.getThriftConnectionUri(), is("thrift://localhost:" + hive.getThriftPort()));
+    HiveConf conf = new HiveConf(ThriftHiveMetaStoreJUnitRuleTest.class);
+    conf.setVar(HiveConf.ConfVars.METASTOREURIS, hive.getThriftConnectionUri());
+    HiveMetaStoreClient client = new HiveMetaStoreClient(conf);
+    List<String> databases = client.getAllDatabases();
+    assertThat(databases.size(), is(2));
+    assertThat(databases.get(0), is("default"));
+    assertThat(databases.get(1), is(databaseName));
+  }
+
+  @Test
+  public void customProperties() {
+    Map<String, String> conf = new HashMap<>();
+    conf.put("my.custom.key", "my.custom.value");
+    HiveConf hiveConf = new ThriftHiveMetaStoreJUnitRule("db", conf).conf();
+    assertThat(hiveConf.get("my.custom.key"), is("my.custom.value"));
+  }
+
+  @Test
+  public void createExistingDatabase() {
+    Assertions.assertThrows(AlreadyExistsException.class, () -> {
```

Comment:
The lambdas from here and the next two tests can also be on one line each :)

Reply:
Maybe except for this one, because it doesn't fit on one line, so it would be something like this
```java
    Assertions.assertThrows(AlreadyExistsException.class,
        () -> hiveDefaultName.createDatabase(hiveDefaultName.databaseName()));
```